### PR TITLE
typing: regression

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
the same also here, most of the `mypy` issues are about
```
error: Signature of "update" incompatible with supertype "Metric"  [override]
```

File: torchmetrics/image/psnr.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def __init__(
         self.reduction = reduction
         self.dim = tuple(dim) if isinstance(dim, Sequence) else dim
 
-    def update(self, preds: Tensor, target: Tensor) -> None:
+    def update(self, preds: Tensor, target: Tensor) -> None:  # type: ignore
```
Comment (justusschock) [inline torchmetrics/image/psnr.py:113]: this is the same as in #332 right? So contradicting to what I wrote there maybe merge them and do a PR with a followup on all signatures if we found a solution?

File: torchmetrics/regression/explained_variance.py [hunk 2]
```diff
@@ -113,11 +113,11 @@ def update(self, preds: Tensor, target: Tensor) -> None:
...
-        self.n_obs = self.n_obs + n_obs
-        self.sum_error = self.sum_error + sum_error
-        self.sum_squared_error = self.sum_squared_error + sum_squared_error
-        self.sum_target = self.sum_target + sum_target
-        self.sum_squared_target = self.sum_squared_target + sum_squared_target
+        self.n_obs: Tensor = self.n_obs + n_obs
+        self.sum_error: Tensor = self.sum_error + sum_error
+        self.sum_squared_error: Tensor = self.sum_squared_error + sum_squared_error
+        self.sum_target: Tensor = self.sum_target + sum_target
+        self.sum_squared_target: Tensor = self.sum_squared_target + sum_squared_target
```
Comment (justusschock) [inline torchmetrics/regression/explained_variance.py:120]: it's better to add the typing already outside (where you usually define static attributes) and not initialize it then typing lately here. 

So something like 

```python

class MyMetric:
    a: torch.Tensor
    def __init__(self):
        super().__init__()
        self.add_state('a', default=torch.tensor(0.))
```

File: torchmetrics/image/psnr.py (hunk 1 above)
Comment (Borda) [reply torchmetrics/image/psnr.py:113]: yes, want to make another PR to try to resolve this typing over all codebase

**Review by justusschock — APPROVED** [review]

**Pull request merged**